### PR TITLE
Crystallisation

### PR DESCRIPTION
**Pull request by seanmtracey**
To the Crystal Dome!

### CODE REVIEW

File: client/js/lib/d3.js [hunk 1]
```diff
@@ -182,11 +177,37 @@ module.exports = function ({
...
+
+		const rev = ringColors.slice(crystallisationIndex + 1, ringColors.length).reverse();
+		for(let i = crystallisationIndex + 1, j = 0; i < ringColors.length; i += 1, j += 1){
+			ringColors[i] = rev[j];
+		}
+
+		rings.forEach((ring, idx) => {
+			ring.fill = ringColors[idx];
+		});
+	}
```
Comment (AdaRoseCannon) [inline client/js/lib/d3.js]: Please can you do the colour changes in main.js to keep all the colour code together.

File: client/js/lib/d3.js [hunk 2]
```diff
@@ -182,11 +177,37 @@ module.exports = function ({
...
+
 	for (let i=0; i<rings.length; i++) {
+
 		rootNode.append('circle')
 			.attr('class', 'background')
 			.attr('r', (rings[i].max + 1) * ringSize)
 			.style('fill', rings[i].fill);
+
+		console.log(rings[i]);
```
Comment (AdaRoseCannon) [inline client/js/lib/d3.js]: Log.

File: client/js/main.js [hunk 3]
```diff
@@ -71,6 +72,7 @@ parseOptions((function () {
 	parsed.showcol = parsed.showcol || '';
 	parsed.sortcol = (parsed.sortcol || 'phase').toLowerCase();
 	parsed.sortcolorder = (parsed.sortcolorder || '');
+	parsed.crystallisation = parsed.crystallisation || null;
```
Comment (AdaRoseCannon) [inline client/js/main.js]: don't need this it is fine for crystallisation to be undefined.

undefined inputs won't get added to the options object.

File: client/js/lib/d3.js (hunk 1 above)
Comment (seanmtracey) [inline client/js/lib/d3.js]: Sure, that makes sense.

**Comment by AdaRoseCannon** [comment]
screenshot 2016-03-16 at 10 31 17

**Pull request merged**